rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -207,30 +210,72 @@ void user_manager_dialog::GenerateUser(const std::string& username)
 	fs::write_file(dev_hdd0 + "home/" + nextUserId + "/localusername", fs::create + fs::excl + fs::write, username);
 }
 
-void user_manager_dialog::OnUserCreate()
+bool user_manager_dialog::ValidateUsername(QString textToValidate)
+{
+	// "Entire string (^...$) must be between 3 and 16 characters
+	// and only consist of letters, numbers, underscores, and hyphens."
+	QRegExpValidator validator(QRegExp("^[A-Za-z0-9_-]{3,16}$"));
+
+	int pos = 0;
+	return (validator.validate(textToValidate, pos) == QValidator::Acceptable);
+}
+
+void user_manager_dialog::OnUserRename()
 {
-	user_create_dialog* createDialog = new user_create_dialog(m_dir, this);
+	int idx = m_table->currentRow();
+	if (idx < 0)
+	{
+		return;
+	}
+	int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+	if (m_user_list[idx_real]->GetUserId() == Emu.GetUsr())
+	{
+		QToolTip::showText(QCursor::pos(), tr("You cannot rename the logged in user!"));
+		return;
+	}
+	user_create_dialog* createDialog = new user_create_dialog("Rename User", this);
 	createDialog->setModal(true);
-	connect(createDialog->GetOkButton(), &QAbstractButton::clicked, this, [=]
+	createDialog->setAttribute(Qt::WA_DeleteOnClose, true);
+	connect(createDialog, &user_create_dialog::OnOkButtonClick, this, [=]
 	{
-		// "Entire string (^...$) must be between 3 and 16 characters
-		// and only consist of letters, numbers, underscores, and hyphens."
-		QRegExpValidator* validator = new QRegExpValidator(QRegExp("^[A-Za-z0-9_-]{3,16}$"));
-
-		int pos = 0;
-		QString textToValidate = createDialog->GetText();
-		if (validator->validate(textToValidate, pos) != QValidator::Acceptable)
+		int idx = m_table->currentRow();
+		if (idx < 0)
 		{
-			QToolTip::showText(QCursor::pos(), tr("Username must have between 3 and 16 characters \nand only contain letters, numbers, underscores, and dashes."));
 			return;
 		}
-		user_manager_dialog::GenerateUser(textToValidate.toStdString());
-		user_manager_dialog::UpdateTable();
-		createDialog->close();
+		int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+		QString textToValidate = createDialog->GetText();
+		if (ValidateUsername(textToValidate))
+		{
+			const std::string dev_hdd0 = Emu.GetHddDir();
+			const std::string userId = m_user_list[idx_real]->GetUserId();
+			fs::remove_file(dev_hdd0 + "home/" + userId + "/localusername");
+			fs::write_file(dev_hdd0 + "home/" + userId + "/localusername", fs::create + fs::excl + fs::write, textToValidate.toStdString());
```

Comment:
I think these can be done in just one step. with an additional fs::trunc.

also don't repeat these concatenations on the other parts and use a string instead

Reply:
Can you explain how trunc would work here? Wouldn't it just cut the name down?

Reply:
Disregard my last comment.